### ISSUE TITLE
About optimizer.step()

### ISSUE THREAD

**Issue by zhouweii234** [hkchengrex/STCN#44]
The code report warnings：warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. 
Since I modified part of your network, I want to make sure I don't misunderstand your code. Does `self.scaler.step(self.optimizer)`  (model/model.py line 163)  means `optimizer.step()`in the code?

![7f092c665e7e2cf16861f75f3635e01](https://user-images.githubusercontent.com/62086490/129880271-9041f7c2-3fd9-451d-a47d-5cfbf628a9b4.png)

**Comment by hkchengrex** [comment]
It doesn't matter.
See this thread: https://discuss.pytorch.org/t/optimizer-step-before-lr-scheduler-step-error-using-gradscaler/92930/3

**Comment by FlyDre** [comment]
> The code report warnings：warnings.warn("Detected call of `lr_scheduler.step()` before `optimizer.step()`. Since I modified part of your network, I want to make sure I don't misunderstand your code. Does `self.scaler.step(self.optimizer)` (model/model.py line 163) means `optimizer.step()`in the code?
> 
> ![7f092c665e7e2cf16861f75f3635e01](https://user-images.githubusercontent.com/62086490/129880271-9041f7c2-3fd9-451d-a47d-5cfbf628a9b4.png)

well, I encounter the same problem ( change ResNet50 to ConvNeXt), and loss1 & loss2 & total_loss turns out to be NaN, could you please show me how you fix it?
